### DISCUSSION TITLE
记录一次重启按钮导致的故障

### DISCUSSION THREAD

**Discussion by seewhyme** [labring/laf#417] [Feedbacks] [answered]
# 问题现象
在依赖管理里加完包以后，点了重启按钮， 导致所有接口502。

看了日志，发现是因为重启会重建docker容器 app-xxx，卡在了 `[INFO] server - run command:  npm install` 这一步，即网络问题无法安装npm包。

# 解决方案
1. 进入应用容器，`docker exec -u root -it fbf42c979ee9 /bin/sh   `。
2. 手动设置  registry 。 `npm config set registry https://mirrors.huaweicloud.com/repository/npm/`
3. 手动install 。  `npm install xxx@latest  --no-audit --no-fund`
4. exit 后， 执行重启 `docker restart fbf42c979ee9` 

# 建议
1. 是否可以支持设置npm 源地址， 让用户自己能设置npm registry，防止重启时依赖无法安装的问题。
2. 依赖管理增加是否安装成功的状态，出问题时好定位。

**Comment by maslow** [comment, marked as answer]
```bash
# ...
instance-controller:
    image: lafyun/instance-controller:latest
    environment:
      APP_SERVICE_ENV_NPM_INSTALL_FLAGS: ' --no-audit --no-fund'
# ...
```

`APP_SERVICE_ENV_NPM_INSTALL_FLAGS`  可以添加 npm install 的参数：`--registry https://mirrors.huaweicloud.com/repository/npm/`
